plugins/modules/ec2_instance.py

```diff
@@ -1284,122 +1407,164 @@ def add_or_update_instance_profile(instance, desired_profile_name):
     return False
 
 
-def build_network_spec(params):
+def validate_assign_public_ip(params: Dict[str, Any]) -> None:
     """
-    Returns list of interfaces [complex]
-    Interface type: {
-        'AssociatePublicIpAddress': True|False,
-        'DeleteOnTermination': True|False,
-        'Description': 'string',
-        'DeviceIndex': 123,
-        'Groups': [
-            'string',
-        ],
-        'Ipv6AddressCount': 123,
-        'Ipv6Addresses': [
-            {
-                'Ipv6Address': 'string'
-            },
-        ],
-        'NetworkInterfaceId': 'string',
-        'PrivateIpAddress': 'string',
-        'PrivateIpAddresses': [
-            {
-                'Primary': True|False,
-                'PrivateIpAddress': 'string'
-            },
-        ],
-        'SecondaryPrivateIpAddressCount': 123,
-        'SubnetId': 'string'
-    },
+    Validate Network interface public IP configuration
+        Parameters:
+            params: module parameters.
     """
+    network_interfaces = params.get("network_interfaces") or []
+    network_interfaces_ids = params.get("network_interfaces_ids") or []
```

Comment:
```suggestion
    network_interfaces = params.get("network_interfaces", [])
    network_interfaces_ids = params.get("network_interfaces_ids", [])
```

Reply:
This will not work as expected. By default ansible set non provided parameters to `None`, therefore using `params.get("network_interface_ids", [])` will produce value `None`